tests/foreman/cli/test_repository.py

```diff
@@ -1865,16 +1865,77 @@ def test_positive_upload_content_srpm(self):
             local_file=get_data_file(SRPM_TO_UPLOAD),
             remote_file="/tmp/{0}".format(SRPM_TO_UPLOAD)
         )
+        # Upload SRPM
         result = Repository.upload_content({
             'name': new_repo['name'],
             'organization': new_repo['organization'],
             'path': "/tmp/{0}".format(SRPM_TO_UPLOAD),
             'product-id': new_repo['product']['id'],
+            'content-type': 'srpm',
         })
-        self.assertIn(
-            "Successfully uploaded file '{0}'".format(SRPM_TO_UPLOAD),
-            result[0]['message'],
+        assert "Successfully uploaded file '{0}'".format(SRPM_TO_UPLOAD) in result[0]['message']
+        assert int('{}'.format(Repository.info({'id': new_repo['id']})
+                               ['content-counts']['source-rpms'])) == 1
+
+        # Remove uploaded SRPM
+        Repository.remove_content({
+            'id': new_repo['id'],
+            'ids': [Srpm.list()[0]['id']],
+            'content-type': 'srpm'
+        })
+        assert int('{}'.format(Repository.info({'id': new_repo['id']})
+                               ['content-counts']['source-rpms'])) == 0
+
+    @tier2
+    def test_positive_srpm_list_end_to_end(self):
+        """Create repository,  upload, list and remove a SRPM content
+
+        :id: 98ad4228-f2e5-438a-9210-5ce6561769f2
+
+        :expectedresults:
+            1. SRPM should be listed repository wise.
+            2. SRPM should be listed product wise.
+            3. SRPM should be listed for specific and all Organizations.
+            4. SRPM should be listed LCE wise.
+            5. Able to see info of uploaded SRPM.
+
+        :CaseImportance: High
+        """
+        new_repo = self._make_repository({'name': gen_string('alpha', 15)})
+        ssh.upload_file(
+            local_file=get_data_file(SRPM_TO_UPLOAD),
+            remote_file="/tmp/{0}".format(SRPM_TO_UPLOAD)
         )
+        # Upload SRPM
+        Repository.upload_content({
+            'name': new_repo['name'],
+            'organization': new_repo['organization'],
+            'path': "/tmp/{0}".format(SRPM_TO_UPLOAD),
+            'product-id': new_repo['product']['id'],
+            'content-type': 'srpm',
+        })
+        srpm_list = Srpm.list()
+        assert srpm_list[0]['filename'] == SRPM_TO_UPLOAD
+        assert len(srpm_list) == 1
+        assert Srpm.info({'id': int(srpm_list[0]['id'])})[0]['filename'] == SRPM_TO_UPLOAD
+        assert int('{}'.format(Repository.info(
+            {'id': new_repo['id']})['content-counts']['source-rpms'])) == len(
+            Srpm.list({'repository-id': new_repo['id']}))
+        assert len(Srpm.list({'organization': new_repo['organization'], 'product-id': new_repo[
+            'product']['id']})) == 1
+        assert len(Srpm.list({'organization': new_repo['organization']})) == 1
```

Comment:
This will look for SRPMs in default organization, so assumption that SRPM created in this test will be the only one seems a little too strong / optimistic for me. It's enough for another test to fail with cleanup for this to fail.

Reply:
New org is creating as part of `Setup` hence using new_repo org details to filter SRPMs org wise .

Reply:
`new_repo` is created using [`self._make_repository`](https://github.com/SatelliteQE/robottelo/blob/master/tests/foreman/cli/test_repository.py#L117-L125), which does not define organization at all. As far as I know, command like that will assume default organization unless explicitly stated.

You would need something like
```python
new_repo = self._make_repository({
    'name': gen_string('alpha', 15),
    'organization-id': self.org['id']
})
```

Reply:
new_org is taking care under `setup`  ..
https://github.com/SatelliteQE/robottelo/blob/f3162b1ddf4eb271b47c85ad3d1574bd8f5654c1/tests/foreman/cli/test_repository.py#L106-L116

Reply:
@vijay8451 You are right, this is tied to new organization through product, created at setUp phase. I see it now. Sorry for being stubborn.